src/sys/unix/uds/stream.rs

```diff
@@ -13,7 +13,7 @@ pub(crate) fn connect(path: &Path) -> io::Result<net::UnixStream> {
 
     match syscall!(connect(socket, sockaddr, socklen)) {
         Ok(_) => {}
-        Err(ref e) if e.kind() == io::ErrorKind::WouldBlock => {}
+        Err(ref err) if err.raw_os_error() == Some(libc::EINPROGRESS) => {}
```

Comment:
I've added `EINPROGRESS` instead because I can't really tell from the macOS manual if they guarantee that `EAGAIN` is always returned (unlike e.g. Linux).

Reply:
Unfortunately I can't tell either; that manual isn't great.